feat: dotnet profiling docs (#8862)
getsentry/sentry-docs: src/platforms/dotnet/profiling/index.mdx

---
title: Set Up Profiling
description: "Learn how to enable profiling in your app if it is not already set up."
sidebar_order: 5000
---

<PlatformContent includePath="profiling/index/preface" />

With [profiling](/product/profiling/), Sentry allows you to collect and analyze performance profiles from real user devices in production to give you a complete picture of how your application performs in a variety of environments.

<Note>

> .NET profiling alpha is available .NET 6.0+ (tested on .NET 7.0 & .NET 8.0 as well) for:

Reviewer: ```suggestion
Sentry profiling for .NET is available in _Alpha_ on .NET 6.0+ (tested on .NET 7.0 & .NET 8.0 as well) for:
```

- Windows
- Linux
- macOS
- iOS

> - macCatalyst

Reviewer: [macOS](https://www.apple.com/ca/macos/sonoma/), [Mac](https://www.apple.com/ca/mac/), [Mac Catalyst](https://developer.apple.com/mac-catalyst/)

```suggestion
- Mac Catalyst
```

</Note>

## Enable Profiling

To enable profiling, set the `ProfilesSampleRate`. 

> Additionally, for all platforms except iOS/Mac Catalyst, you need to add a dependency on the `Sentry.Profiling` NuGet package.

Reviewer: ```suggestion
Additionally, for all platforms except iOS/Mac Catalyst, you need to add a dependency on the `Sentry.Profiling` NuGet package.

```shell {tabTitle:.NET CLI}
dotnet add package Sentry.Profiling -v {{@inject packages.version('sentry.dotnet', '4.0.0-beta.8') }}
```
```

Author: P. S. you need to use four backticks in order to use 3 in the actual suggestion

````suggestion
Additionally, for all platforms except iOS/Mac Catalyst, you need to add a dependency on the `Sentry.Profiling` NuGet package.
	
	```shell {tabTitle:.NET CLI}
	dotnet add package Sentry.Profiling -v {{@inject packages.version('sentry.dotnet', '4.0.0-beta.8') }}
	```
````

Profiling depends on Sentry’s performance monitoring product being enabled beforehand. To enable performance monitoring in the SDK, set the `TracesSampleRate` option to the desired value.

> ```csharp {tabTitle:iOS/macCatalyst}

Reviewer: Also, please invert the order. Show WIn/Mac/Linux first, iOS/MacCatalyst after

````suggestion
```csharp {tabTitle:iOS/MacCatalyst}
````

> using Sentry;
> 
> SentrySdk.Init(options =>

Reviewer: From the SDK 4.0 it'll be on global usings: https://github.com/getsentry/sentry-dotnet/pull/3043

```suggestion
SentrySdk.Init(options =>
```

{
    // ... usual setup options omitted for clarity (see Getting Started) ...
    
    // Sample rate for your transactions, e.g. value 0.1 means we want to report 10% of transactions. 1.0 is 100%
    options.TracesSampleRate = 1.0;
    
    // Sample rate for profiling, applied on top of othe TracesSampleRate, 
    // e.g. 0.2 means we want to profile 20 % of the captured transactions, i.e. 2 % of all transactions.
    options.ProfilesSampleRate = 0.2;
});
```

```csharp {tabTitle:Windows/Linux/macOS}

> using Sentry;
> 

Reviewer: ```suggestion
```

SentrySdk.Init(options =>
{
    // ... usual setup options omitted for clarity (see Getting Started) ...
    
    // Sample rate for your transactions, e.g. value 0.1 means we want to report 10% of transactions.

>     options.TracesSampleRate = 0.1;

Reviewer: Align with the snippet and comment above, lets go with 100% here mentioning folks shoudl update this in production. It's frustraiting to try it out if it doesn't work on each attempt

```suggestion
    options.TracesSampleRate = 1.0;
```

>     // Sample rate for your transactions, e.g. value 0.1 means we want to report 10% of transactions.
>     options.TracesSampleRate = 0.1;

Author: ```suggestion
    // Sample rate for your transactions, e.g. value 0.1 means we want to report 10% of transactions.
    // Setting 1.0 means all transactions are profiled. You should update this to fit your app for production use.
    options.TracesSampleRate = 0.1;
```

    
    // Sample rate for profiling, applied on top of othe TracesSampleRate, 
    // e.g. 0.2 means we want to profile 20 % of the captured transactions, i.e. 2 % of all transactions.
    options.ProfilesSampleRate = 0.2;
    
    // Requires NuGet package: Sentry.Profiling
    // Note: by default, the profiler is initialized asynchronously, this can be tuned by passing an desired initialization timeout to the constructor.

>     options.AddIntegration(new ProfilingIntegration());

Reviewer: How about we show the code that blocks to show how it works?
```suggestion
    options.AddIntegration(new ProfilingIntegration(
        // During startup, wait up to 500ms to profile the app startup code. This could make launching the app a bit slower so comment it out if your prefer profiling to start asynchronously
        TimeSpan.FromMilliseconds(500)
    ));
```

---
title: Set Up Profiling
description: "Learn how to enable profiling in your app if it is not already set up."
sidebar_order: 5000
---

<PlatformContent includePath="profiling/index/preface" />

With [profiling](/product/profiling/), Sentry allows you to collect and analyze performance profiles from real user devices in production to give you a complete picture of how your application performs in a variety of environments.

<Note>

.NET profiling alpha is available for:

> - .NET 6.0 and above for Windows, Linux and macOS using [.NET Core CLR Diagnostics](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/diagnostics-client-library)

Reviewer: This is good context, but better to keep it for some detailed docs further down below. Here it's just too much context

> using [.NET Core CLR Diagnostics](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/diagnostics-client-library)

```suggestion
- .NET 6.0 and above for Windows, Linux and macOS
```

> - .NET 6.0 and above for iOS and macCatalyst using native profiling support in [Sentry Apple SDK](/platforms/apple/)

Reviewer: ```suggestion
- .NET 6.0 and above for iOS and Mac Catalyst using native profiling support in [Sentry Apple SDK](/platforms/apple/)
```

.NET profiling is currently **not** available for:

> - NativeAOT

Reviewer: ```suggestion
- Native AOT (except on iOS)
```

- .NET Framework
- Android

</Note>

> ## Enable Performance Monitoring
> 
> Profiling depends on Sentry’s performance monitoring product being enabled beforehand. To enable performance monitoring in the SDK, set the `TracesSampleRate` option to the desired value.
> 
> ```csharp
> using Sentry;
> 
> SentrySdk.Init(options =>
> {
>     // ... usual setup options omitted for clarity (see Getting Started) ...
>     
>     // Sample rate for your transactions, e.g. value 0.1 means we want to report 10% of transactions.
>     options.TracesSampleRate = 0.1;
> });
> ```
> 
> ## Enable Profiling
> 
> To enable profiling, set the `ProfilesSampleRate`. 
> Additionally, for all platforms except iOS/macCatalyst, you need to add a dependency on the `Sentry.Profiling` NuGet package.
> 
> ```csharp {tabTitle:iOS/macCatalyst}
> using Sentry;
> 
> SentrySdk.Init(options =>
> {
>     // ... usual setup options omitted for clarity (see Getting Started) ...
>     
>     // Sample rate for your transactions, e.g. value 0.1 means we want to report 10% of transactions.
>     options.TracesSampleRate = 0.1;
>     
>     // Sample rate for profiling, applied on top of othe TracesSampleRate, 
>     // e.g. 0.2 means we want to profile 20 % of the captured transactions, i.e. 2 % of all transactions.
>     options.ProfilesSampleRate = 0.2;
> });
> ```

Reviewer: ````suggestion
## Enable Profiling

To enable profiling, set the `ProfilesSampleRate`. 
Additionally, for all platforms except iOS/Mac Catalyst, you need to add a dependency on the `Sentry.Profiling` NuGet package.

Profiling depends on Sentry’s performance monitoring product being enabled beforehand. To enable performance monitoring in the SDK, set the `TracesSampleRate` option to the desired value.

```csharp {tabTitle:iOS/macCatalyst}
using Sentry;

SentrySdk.Init(options =>
{
    // ... usual setup options omitted for clarity (see Getting Started) ...
    
    // Sample rate for your transactions, e.g. value 0.1 means we want to report 10% of transactions. 1.0 is 100%
    options.TracesSampleRate = 1.0;
    
    // Sample rate for profiling, applied on top of othe TracesSampleRate, 
    // e.g. 0.2 means we want to profile 20 % of the captured transactions, i.e. 2 % of all transactions.
    options.ProfilesSampleRate = 0.2;
});
```
````
